src/algorithms/generate.rs

```diff
@@ -13,10 +11,10 @@ use crate::{
 };
 
 pub struct RsaPrivateKeyComponents {
-    pub n: BigUint,
-    pub e: BigUint,
-    pub d: BigUint,
-    pub primes: Vec<BigUint>,
+    pub n: Odd<BoxedUint>,
+    pub e: u64,
```

Comment:
I don't think it's good to limit `e` to 64 bits. See [RFC 8017 Section 3.1](https://datatracker.ietf.org/doc/html/rfc8017#section-3.1):

> In a valid RSA public key, the RSA modulus n is a product of u distinct odd primes r_i, i = 1, 2, ..., u, where u >= 2, and the RSA public exponent e is an integer between 3 and n - 1 satisfying GCD(e,\lambda(n)) = 1, where \lambda(n) = LCM(r_1 - 1, ..., r_u - 1). By convention, the first two primes r_1 and r_2 may also be denoted p and q, respectively.

And [FIPS 186-4 page 52](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf#page=62)

> The exponent e **shall** be an odd positive integer such that:
>        2¹⁶ < e < 2²⁵⁶.

It should still be `BoxedUint`.

Reply:
thanks for pointing this out, our default safe constants restricting it, but you are right, it should be possible for larger exponents. this is now fixed